metricbeat/module/docker/cpu/helper.go

```diff
@@ -78,21 +78,25 @@ func perCpuUsage(stats *dc.Stats) common.MapStr {
 // TODO: These helper should be merged with the cpu helper in system/cpu
 
 func totalUsage(stats *dc.Stats) float64 {
-	return calculateLoad(stats.CPUStats.CPUUsage.TotalUsage - stats.PreCPUStats.CPUUsage.TotalUsage)
+	return calculateLoad(stats.CPUStats.CPUUsage.TotalUsage, stats.PreCPUStats.CPUUsage.TotalUsage)
 }
 
 func usageInKernelmode(stats *dc.Stats) float64 {
-	return calculateLoad(stats.CPUStats.CPUUsage.UsageInKernelmode - stats.PreCPUStats.CPUUsage.UsageInKernelmode)
+	return calculateLoad(stats.CPUStats.CPUUsage.UsageInKernelmode, stats.PreCPUStats.CPUUsage.UsageInKernelmode)
 }
 
 func usageInUsermode(stats *dc.Stats) float64 {
-	return calculateLoad(stats.CPUStats.CPUUsage.UsageInUsermode - stats.PreCPUStats.CPUUsage.UsageInUsermode)
+	return calculateLoad(stats.CPUStats.CPUUsage.UsageInUsermode, stats.PreCPUStats.CPUUsage.UsageInUsermode)
 }
 
 func systemUsage(stats *dc.Stats) float64 {
-	return calculateLoad(stats.CPUStats.SystemCPUUsage - stats.PreCPUStats.SystemCPUUsage)
+	return calculateLoad(stats.CPUStats.SystemCPUUsage, stats.PreCPUStats.SystemCPUUsage)
 }
 
-func calculateLoad(value uint64) float64 {
-	return float64(value) / float64(1000000000)
+func calculateLoad(newValue uint64, oldValue uint64) float64 {
+	value := float64(newValue) - float64(oldValue)
```

Comment:
This function is meant to calculate the % CPU time change between two successive readings. The "oldVlue" represents the preCpu which refers to the  CPU statistics of the last read. 
Time here is expressed by second and not by milliseconds. ( The main goal is to expose the %, in the same way, it's displayed by docker Client)